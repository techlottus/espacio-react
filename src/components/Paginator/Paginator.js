import React, { useEffect } from "react";

export const Paginator = ({ data = {} }) => {
  const paginatorRef = React.createRef();

  useEffect(() => {
    paginatorRef.current.data = {
      iconPrevious: data.iconPrevious || "",
      iconNext: data.iconNext || "",
      size: data.size || "",
      id: data.id || "",
      maxNumbers: data.maxNumbers || 0,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-paginator ref={paginatorRef}></lottus-paginator>
    </>
  );
};
