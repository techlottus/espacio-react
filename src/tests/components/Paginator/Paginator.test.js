import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Paginator } from "../../../components/Paginator/Paginator";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let data = {
    iconPrevious: "",
    iconNext: "",
    size: "",
    maxNumbers: 0,
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Paginator data={data} />
    </Provider>
  );
};

describe("Test <Paginator/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
