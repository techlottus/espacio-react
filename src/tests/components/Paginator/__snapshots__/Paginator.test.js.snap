// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Paginator/> should init component 1`] = `<lottus-paginator />`;
